frontend/src/modules/SeismicIntersection/view.tsx

```diff
@@ -0,0 +1,215 @@
+import { ContentError } from "@modules/_shared/components/ContentMessage";
+
+import { isEqual } from "lodash";
+
+import { useSeismicFenceDataQuery } from "./queryHooks";
+import { State } from "./state";
+import { addMDOverlay, addSeismicLayer, addWellborePathLayer } from "./utils/esvIntersectionControllerUtils";
+import {
+    createSeismicSliceImageDataArrayFromFenceData,
+    createSeismicSliceImageYAxisValuesArrayFromFenceData,
+    makeExtendedTrajectoryFromTrajectoryXyzPoints,
+    makeReferenceSystemFromTrajectoryXyzPoints,
+    makeTrajectoryXyzPointsFromWellboreTrajectory,
+} from "./utils/esvIntersectionDataConversion";
+import {
+    SeismicSliceImageOptions,
+    SeismicSliceImageStatus,
+    useGenerateSeismicSliceImage,
+} from "./utils/esvIntersectionHooks";
+
+export const view = ({ moduleContext, workbenchSettings }: ModuleFCProps<State>) => {
+    const wrapperDivRef = React.useRef<HTMLDivElement | null>(null);
+    const wrapperDivSize = useElementSize(wrapperDivRef);
+    const esvIntersectionContainerRef = React.useRef<HTMLDivElement | null>(null);
+    const esvIntersectionControllerRef = React.useRef<Controller | null>(null);
+
+    const gridLayerUuid = useId();
+    const statusWriter = useViewStatusWriter(moduleContext);
+
+    const seismicAddress = moduleContext.useStoreValue("seismicAddress");
+    const wellboreAddress = moduleContext.useStoreValue("wellboreAddress");
+    const extension = moduleContext.useStoreValue("extension");
+    const zScale = moduleContext.useStoreValue("zScale");
+
+    const seismicColorScale = workbenchSettings.useDiscreteColorScale({
+        gradientType: ColorScaleGradientType.Diverging,
+    });
+
+    const [seismicColors, setSeismicColors] = React.useState<string[]>(seismicColorScale.getColorPalette().getColors());
+    if (!isEqual(seismicColorScale.getColorPalette().getColors(), seismicColors)) {
+        setSeismicColors(seismicColorScale.getColorPalette().getColors());
+    }
+
+    // Extended wellbore trajectory for creating intersection/fence extended on both sides of wellbore
+    const [extendedWellboreTrajectory, setExtendedWellboreTrajectory] = React.useState<Trajectory | null>(null);
+
+    // Array of 3D points [x,y,z] for well trajectory layer in esv-intersection (to be in synch with seismic fence layer)
+    const [renderWellboreTrajectoryXyzPoints, setRenderWellboreTrajectoryXyzPoints] = React.useState<number[][] | null>(
+        null
+    );
+
+    // Data for seismic fence layer in esv-intersection
+    const [seismicFencePolyline, setSeismicFencePolyline] = React.useState<SeismicFencePolyline_api | null>(null);
+
+    // Async generating seismic slice image
+    const [generateSeismicSliceImageOptions, setGenerateSeismicSliceImageOptions] =
+        React.useState<SeismicSliceImageOptions | null>(null);
+    const generateSeismicSliceImageHook = useGenerateSeismicSliceImage(generateSeismicSliceImageOptions);
```

Comment:
I wouldn't use the word hook here. A hook is just an instrument, the data you get returned should not have the name of the instrument used in it. Use a word that reflects the returned data object.